core/src/main/scala/dagr/core/tasksystem/EitherTask.scala

```diff
@@ -23,24 +23,31 @@
  */
 package dagr.core.tasksystem
 
-/** Helps determine which task to perform for an [[EitherTask]] */
-object EitherTaskFlag extends Enumeration {
-  type EitherTaskFlag = Value
-  val LEFT, RIGHT = Value
-}
-
 object EitherTask {
+  sealed trait Choice
+  object Left  extends Choice
+  object Right extends Choice
+
+  /**
+    * Creates an [[EitherTask]] that wraps `choice` into a function that will be evaluated when the
+    * [[EitherTask]] is evaluated at execution time.
```

Comment:
not necessarily true, since it is evaluated prior to execution time.  It is evaluated when it has no dependencies.